src/olympia/blocklist/tests/test_mlbf.py

```diff
@@ -452,6 +452,145 @@ def test_diff_block_soft_to_hard(self):
             ),
         }
 
+    def test_diff_all_possible_changes(self):
+        """
+        Simulates 6 guid:version combinations moving from one state to another
+        covering all possible movements and the final diff state.
+
+        1. Not blocked -> Soft blocked
+        2. Not blocked -> Blocked
+        3. Soft blocked -> Blocked
+        4. Soft blocked -> Not blocked
+        5. Blocked -> Soft blocked
+        6. Blocked -> Not blocked
+
+        first stash
+        {
+          'blocked': [5,6],
+          'softblocked': [3,4],
+          # 1,2 are already unblocked so wouldn't be on the stash
```

Comment:
I am not clear on why that's `true`.

Reply:
What what is true? 1 and 2 were blocked before we re-generated the stash so we don't add them to "unblocked".

Reply:
Comments in GitHub diffs are contextual. They apply to individual lines (or multiple lines, sometimes).

> 1 and 2 were blocked before we re-generated the stash so we don't add them to "unblocked".

That wasn't entirely clear to me. I read the comment then jumped to the generation of the MLBF and the assertion, and didn't quite get how we ended in this state. Maybe splitting the long python comment into inlined comments in the test would be easier to follow? I don't know.

Reply:
I know which line you referred to but it wasn't clear to me exactly which part of the statement being true is not clear. Would something like

"Versions that were already not hard or softblocked will not be included in any stash since their state has not changed"

Would that have helped do you think?

Reply:
Not really. I think it's more mapping parts of the large comment to what happens in the test that could be helpful.

Reply:
Updated by removing the long comment and trying to make the inline comments more clear.